presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -769,6 +769,8 @@ public ListeningExecutorService createResourceManagerExecutor(ResourceManagerCon
 
         //Optional Status Detector
         newOptionalBinder(binder, NodeStatusService.class);
+        //Optional worker status notifier
+        newOptionalBinder(binder, NodeStatusNotifier.class);
```

Comment:
In order to use this, one must create a build of Presto.  Should it be a plugin instead?

Reply:
Makes sense. I rewrote it as a plugin, had a good learning experience with plugin creation :)